### ISSUE TITLE
Cannot find type definition on `"module": "NodeNext"`

### ISSUE THREAD

**Issue by dqn** [gvergnaud/ts-pattern#89]
**Describe the bug**

package.json:
```json
{
  "type": "module",
  "license": "MIT",
  "devDependencies": {
    "typescript": "^4.7.2"
  },
  "dependencies": {
    "ts-pattern": "^4.0.2"
  }
}
```

tsconfig.json:
```json
{
  "compilerOptions": {
    "target": "ESNext",
    "module": "NodeNext",
    "strict": true,
    "skipLibCheck": true
  }
}
```

```ts
import { match } from "ts-pattern";
// Could not find a declaration file for module 'ts-pattern'. '/my-project/node_modules/ts-pattern/dist/index.modern.js' implicitly has an 'any' type.
// Try `npm i --save-dev @types/ts-pattern` if it exists or add a new declaration (.d.ts) file containing `declare module 'ts-pattern';`
```

Workarounds:
- Copying node_modules/ts-pattern/dist/index.d.ts to node_modules/ts-pattern/dist/index.modern.d.ts
- Removing `exports` field from node_modules/ts-pattern/package.json

**Versions**
- TypeScript version: 4.7.2
- ts-pattern version: 4.0.2
- environment: Node.js v18.0.0

**Comment by dqn** [comment]
This issue should be resolved by changing package.json as follows

```json
"exports": {
  "require": {
    "default": "./dist/index.cjs",
    "types": "./dist/index.d.ts"
  },
  "default": {
    "default": "./dist/index.modern.js",
    "types": "./dist/index.d.ts"
  }
},
```

or by Microbundle supporting output `.d.mts`.

FYI: https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing

**Comment by gvergnaud** [comment]
Hey! 
Thanks for reporting this issue, I fixed the "exports" definition in v4.0.3
